pkg/kn/commands/service/update.go

```diff
@@ -92,7 +92,7 @@ func NewServiceUpdateCommand(p *commands.KnParams) *cobra.Command {
 				}
 
 				if trafficFlags.Changed(cmd) {
-					traffic, err := traffic.Compute(cmd, service.Spec.Traffic, &trafficFlags)
+					traffic, err := traffic.Compute(cmd, service.Spec.Traffic, &trafficFlags, service.Name)
```

Comment:
Added `serviceName` parameter to `traffic.Compute` to pass service name for error messaging. Let me know if there is another option for getting the service name in `compute.go` that would get rid of the need to pass data from here.